packages/flutter/lib/src/cupertino/switch.dart

```diff
@@ -352,283 +565,445 @@ class _CupertinoSwitchState extends State<CupertinoSwitch> with TickerProviderSt
+      child: GestureDetector(
+        excludeFromSemantics: true,
+        onHorizontalDragStart: _handleDragStart,
+        onHorizontalDragUpdate: _handleDragUpdate,
+        onHorizontalDragEnd: _handleDragEnd,
+        dragStartBehavior: widget.dragStartBehavior,
+        child: Opacity(
+          opacity: onChanged == null ? _kDisabledOpacity : 1,
+          child: buildToggleable(
+            mouseCursor: effectiveMouseCursor,
+            focusNode: widget.focusNode,
+            onFocusChange: widget.onFocusChange,
+            autofocus: widget.autofocus,
+            size: _kSwitchSize,
+            painter: _painter
+              ..position = position
+              ..reaction = reaction
+              ..reactionFocusFade = reactionFocusFade
+              ..reactionHoverFade = reactionHoverFade
+              ..focusColor = CupertinoDynamicColor.resolve(
+                  widget.focusColor ??
+                  HSLColor
+                    .fromColor(activeColor
+                    .withOpacity(_kCupertinoFocusColorOpacity))
+                    .withLightness(_kCupertinoFocusColorBrightness)
+                    .withSaturation(_kCupertinoFocusColorSaturation)
                     .toColor(),
-              context),
-            onChanged: widget.onChanged,
-            textDirection: Directionality.of(context),
-            isFocused: isFocused,
-            state: this,
-            onOffLabelColors: onOffLabelColors,
+                  context)
+              ..downPosition = downPosition
+              ..isFocused = states.contains(WidgetState.focused)
+              ..isHovered = states.contains(WidgetState.hovered)
+              ..activeColor = effectiveActiveThumbColor
+              ..inactiveColor = effectiveInactiveThumbColor
+              ..activePressedColor = effectiveActivePressedThumbColor
+              ..onOffLabelColors = onOffLabelColors
+              ..inactivePressedColor = effectiveInactivePressedThumbColor
+              ..activeThumbImage = widget.activeThumbImage
+              ..onActiveThumbImageError = widget.onActiveThumbImageError
+              ..inactiveThumbImage = widget.inactiveThumbImage
+              ..onInactiveThumbImageError = widget.onInactiveThumbImageError
+              ..activeTrackColor = effectiveActiveTrackColor
+              ..activeTrackOutlineColor = effectiveActiveTrackOutlineColor
+              ..activeTrackOutlineWidth = effectiveActiveTrackOutlineWidth
+              ..inactiveTrackColor = effectiveInactiveTrackColor
+              ..inactiveTrackOutlineColor = effectiveInactiveTrackOutlineColor
+              ..inactiveTrackOutlineWidth = effectiveInactiveTrackOutlineWidth
+              ..configuration = createLocalImageConfiguration(context)
+              ..isInteractive = isInteractive
+              ..trackInnerLength = _trackInnerLength
+              ..textDirection = Directionality.of(context)
+              ..thumbRadius = _kThumbRadius
```

Comment:
If they are just _k*, we can directly use theme in the `_SwitchPainter` class. I think this can make the implementation simpler:)